.test-infra/dataproc/create_flink_cluster.sh

```diff
@@ -19,28 +19,33 @@
 #
 #    Provide the following environment to run this script:
 #
+#    GCLOUD_ZONE: Google cloud zone. Optional. Default: "us-central1-a"
+#    DATAPROC_VERSION: Dataproc version. Optional. Default: 1.2
 #    CLUSTER_NAME: Cluster name
 #    GCS_BUCKET: GCS bucket url for Dataproc resources (init actions)
-#    HARNESS_IMAGES_TO_PULL: Urls to SDK Harness' images to pull on dataproc workers (accepts 1 or more urls)
+#    HARNESS_IMAGES_TO_PULL: Urls to SDK Harness' images to pull on dataproc workers (optional)
+#    JOB_SERVER_IMAGE: Url to job server docker image to pull on dataproc master (optional)
+#    ARTIFACTS_DIR: Url to bucket where artifacts will be stored for staging (optional)
 #    FLINK_DOWNLOAD_URL: Url to Flink .tar archive to be installed on the cluster
 #    FLINK_NUM_WORKERS: Number of Flink workers
-#    TASK_MANAGER_SLOTS: Number of Flink slots per worker
 #    DETACHED_MODE: Detached mode: should the SSH tunnel run in detached mode?
 #
 #    Example usage:
-#
 #    CLUSTER_NAME=flink \
 #    GCS_BUCKET=gs://<GCS_BUCKET>/flink \
 #    HARNESS_IMAGES_TO_PULL='gcr.io/<IMAGE_REPOSITORY>/python:latest gcr.io/<IMAGE_REPOSITORY>/java:latest' \
+#    JOB_SERVER_IMAGE=gcr.io/<IMAGE_REPOSITORY>/job-server-flink:latest \
+#    ARTIFACTS_DIR=gs://<bucket-for-artifacts> \
 #    FLINK_DOWNLOAD_URL=http://archive.apache.org/dist/flink/flink-1.7.0/flink-1.7.0-bin-hadoop28-scala_2.12.tgz \
 #    FLINK_NUM_WORKERS=2 \
-#    TASK_MANAGER_SLOTS=1 \
 #    DETACHED_MODE=false \
 #    ./create_flink_cluster.sh
 #
 set -Eeuxo pipefail
 
-DATAPROC_VERSION=1.2
+# GCloud properties
+GCLOUD_ZONE="${GCLOUD_ZONE:=us-central1-a}"
+DATAPROC_VERSION="${DATAPROC_VERSION:=1.2}"
```

Comment:
We are only compatible with Dataporc Version 1.2
Should we mention it here and file a jira/bug for it?

Reply:
Could you elaborate on why aren't we compatibile? I was under impression that we can use all dataproc versions depending on our usecase. I deliberately left 1.2 because it fits our currently supported flink versions best (hadoop, scala and other underlying deps): https://cloud.google.com/dataproc/docs/concepts/versioning/dataproc-release-1.2

Reply:
Dataproc updated the Hadoop version 1.3 and Flink does not have a binary for that hadoop version causing the incompatibility.
I am not sure if its fixed or Flink has new artifacts with updated hadoop version.

Reply:
I see. Thanks for explaining. I noticed that since flink 1.8.0 releases are be done without hadoop. I think this means that probably we'll have to do some amendments in the script. However, we do not support 1.8 yet (it is 2 days old! :D ) so I won't investigate it now. We can modify the docs/variable if needed in further prs. 

If curious, see: https://issues.apache.org/jira/browse/FLINK-11266 :)